packages/design-system/src/components/Button/Button.tsx

```diff
@@ -1,9 +1,66 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
 
-export class Button extends React.PureComponent {
-  constructor(props) {
+export interface ButtonProps {
+  /**
+   * Label text or HTML
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classes to be added to the root button element.
+   * Useful for adding utility classes.
+   */
+  className?: string;
+  /**
+   * When provided, this will render the passed in component. This is useful when
+   * integrating with React Router's `<Link>` or using your own custom component.
+   */
+  component?: React.ReactElement<any> | any | ((...args: any[]) => any);
+  disabled?: boolean;
+  /**
+   * When provided the root component will render as an `<a>` element
+   * rather than `button`.
+   */
+  href?: string;
+  /**
+   * Access a reference to the `button` or `a` element
+   */
+  inputRef?: (...args: any[]) => any;
+  /**
+   * @hide-prop [Deprecated] Use inversed instead
+   */
+  inverse?: boolean;
+  /** Applies the inverse theme styling */
+  inversed?: boolean;
+  /**
+   * Returns the [`SyntheticEvent`](https://facebook.github.io/react/docs/events.html).
+   * Not called when the button is disabled.
+   */
+  onClick?: (...args: any[]) => any;
+  size?: 'small' | 'big';
+  /**
+   * Button [`type`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type) attribute
+   */
+  type?: 'button' | 'submit';
+  /**
+   * A string corresponding to the button-component variation classes.
+   * The `'danger'` variation is deprecated and will be removed in a future release.
+   */
+  variation?: 'primary' | 'danger' | 'success' | 'transparent';
+}
+
+type OmitProps = 'size' | 'onClick';
```

Comment:
[fix] related to our conversation about Omit... looking at the documentation site, some of the props aren't showing up. It seems like type, disabled, className and children are not showing up in the doc table, and a new prop (ref) is there.

<img width="1542" alt="Screen Shot 2021-07-27 at 10 13 23 AM" src="https://user-images.githubusercontent.com/33579665/127189858-29491249-8015-4399-8e86-2f5646539c19.png">


The missing props can be resolved by changing this line to
`type OmitProps = 'size' | 'onClick' | 'type' | 'disabled' | 'className' | 'children';`
Not sure how or why `ref` is there

Reply:
Good callout! 

I wonder if I need to check the props table when implementing a button link by default as well. I think under the hood, if you provide an `href` it turns `<button>` to an `<a>`, which would make it inherit the `HTMLAnchorElement` interface props instead, possibly introducing even more props to omit. I'm going to test this out to confirm.

Reply:
I resolved this by adding both 'button' and 'a' to the Omit arg and added all missing props from prop table to OmitProps type. Appear to have discovered a bug with how union types are rendered on the docs site and filed a bug to track the work on that.